test/python/docker/__init__.py

```diff
@@ -44,17 +44,27 @@ def __init__(self):
         os.environ["CONTAINERS_REGISTRIES_CONF"] = os.path.join(
             self.anchor_directory, "registry.conf"
         )
-        conf = """unqualified-search-registries = ["docker.io", "quay.io"]
 
-[[registry]]
-location="localhost:5000"
-insecure=true
+        # Entry verified by compat/test_system.py
+        reg_conf_sfx = """
 
 [[registry.mirror]]
 location = "mirror.localhost:5000"
 
 """
 
+        # Assume developer-mode testing by default
+        reg_conf_source_path="./test/registries.conf"
+
+        # When operating in a CI environment, use the local registry server.
+        # Ref: https://github.com/containers/automation_images/pull/357
+        #      https://github.com/containers/podman/pull/22726
+        if os.getenv("CI_USE_REGISTRY_CACHE"):
+            reg_conf_source_path = "./test/registries-cached.conf"
+
+        with open(os.path.join(reg_conf_source_path)) as file:
+            conf = file.read() + reg_conf_sfx
+
```

Comment:
if you feel like it below there is a bunch of CNI references that are not used at all anymore and could be dropped

Reply:
Okay, ya I was scratching my head on that.  Maybe it's best done as a followup PR.

Reply:
agree, I found some other cni references as well so if I find some free time I might get around to nuke that stuff finally.

Reply:
Oh okay, thanks.  I started on https://github.com/containers/podman/pull/23552 but it can't move forward until this merges (b/c of `changesInclude()` updates).  I have no close attachment to that PR if you have a mind to take on this effort, otherwise I'll continue as best I can after this merges.